src/app/public/modules/country-field/country-field.component.ts

```diff
@@ -0,0 +1,295 @@
+    private windowRef: SkyAppWindowRef
+  ) {
+    /**
+     * The json functions here ensures that we get a copy of the array and not the global original.
+     * This ensures that multiple instances of the component don't overwrite the original data.
+     *
+     * We must type the window object as any here as the intl-tel-input library adds its object
+     * to the main window object.
+     */
+    this.countries = JSON.parse(JSON.stringify((window as any)
+      .intlTelInputGlobals.getCountryData()));
+    this.defaultCountryData = this.countries.find(country => country.iso2 === 'us');
+
+    this.countrySearchForm = this.formBuilder.group({
+      countrySearch: new FormControl()
+    });
+
+    if ((<HTMLElement>elRef.nativeElement.parentElement)
+      .classList.contains('sky-phone-field-country-search')) {
+      this.isInPhoneField = true;
+    }
+  }
+
+  public ngOnInit(): void {
+    if (!this.defaultCountry) {
+      this.defaultCountry = 'us';
+    }
+
+    this.countrySearchForm.get('countrySearch').valueChanges.subscribe(newValue => {
+      if (newValue) {
+        this.selectedCountry = newValue;
+      }
+    });
+    this.addEventListeners();
+  }
+
+  public ngOnDestroy(): void {
+    this.selectedCountryChange.complete();
+    this.removeEventListeners();
+  }
+
+  /**
+   * Sets the country to validate against based on the county's iso2 code.
+   * @param countryCode The International Organization for Standardization's two-letter code
+   * for the default country.
+   */
+  public onCountrySelected(newCountry: SkyAutocompleteSelectionChange): void {
+    if (newCountry.selectedItem) {
+      this.writeValue(this.countries.find(countryInfo => countryInfo.iso2 ===
+        newCountry.selectedItem.iso2));
+    } else {
+      this.writeValue(undefined);
+    }
+  }
+
+  // Angular automatically constructs these methods.
+  /* istanbul ignore next */
+  public onChange: Function = (value: SkyCountryFieldCountry) => { };
+
+  /* istanbul ignore next */
+  public onTouched: Function = () => { };
+
+  public registerOnChange(fn: (value: any) => void): void {
+    this.onChange = fn;
+  }
+
+  public registerOnTouched(fn: () => void): void {
+    this.onTouched = fn;
+  }
+
+  // Allows Angular to disable the input.
+  public setDisabledState(disabled: boolean): void {
+    this.removeEventListeners();
+
+    if (!disabled) {
+      this.addEventListeners();
+    }
+
+    this.disabled = disabled;
+    this.changeDetector.markForCheck();
+  }
+
+  public validate(control: AbstractControl): ValidationErrors {
+    if (!this.control) {
+      this.control = control;
+    }
+    return;
+  }
+
+  public writeValue(value: SkyCountryFieldCountry): void {
+    if (!this.disabled) {
+      this.selectedCountry = value;
+    }
+  }
+
+  private addEventListeners(): void {
+    this.idle = new Subject();
+
+    const documentObj = this.windowRef.nativeWindow.document;
+    fromEvent(documentObj, 'mousedown').takeUntil(this.idle)
```

Comment:
Your newline formatting here doesn't match the other subscribes.

Reply:
Fixed